test/testdrive/protobuf-required.td

```diff
@@ -0,0 +1,23 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Test that required Protobuf fields are unsupported.
+
+$ file-append path=required.proto
+syntax = "proto2";
+
+message Required {
+  required int32 f = 1;
+}
+
+$ protobuf-compile-descriptors inputs=required.proto output=required.pb
```

Comment:
nice! I think these files, especially because they're so concise now, are good examples for anyone actually trying to use these types of sources.

e.g. the format of the input FILE below was something I was confused by for a little bit, partly because of the magic in testdrive.  This "explicitly compile the proto and then use it in the `CREATE SOURCE` statement" flow is much more clear!